frontend/src/Elements/components/Objection/Objection.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Switch from '../Common/Switch';
+
+class Objection extends React.PureComponent {
+  state = {
+    allowDataProcessing: true,
+    busy: true
+  };
+  async componentDidMount() {
+    const { objection } = await this.props.cg.Subject.getObjectionStatus();
+    this.setState({ busy: false, allowDataProcessing: !objection });
+  }
+  async toggleDataProcessingAgreement() {
+    const { busy, allowDataProcessing } = this.state;
+    if (busy) {
+      return;
+    }
+
+    this.setState({ allowDataProcessing: !allowDataProcessing, busy: true });
+    await this.props.cg.Subject.updateObjection(allowDataProcessing);
+    this.setState({ busy: false });
+  }
+  render() {
+    const { label = 'Object data processing' } = this.props.options;
+    const { allowDataProcessing, busy } = this.state;
+
+    return (
+      <div style={{ alignItems: 'center', display: 'flex' }}>
+        <label>{label}</label>&nbsp;
+        <Switch
+          onChange={this.toggleDataProcessingAgreement.bind(this)}
```

Comment:
Just noticed this, if you want to avoid the `bind(this)` stuff, you can declare the function as a class property, that with the help of Babel it gets compiled automatically. We're using that syntax in the `admin` but can be done here as well.

Reply:
Thanks, I'll check.